test/e2e/scheduling/device-plugin.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("[Feature:GPU]", func() {
```

Comment:
I kind of against this being a e2e unless it can be more generally applied across (N) environments.

Reply:
This is the current behavior (and has been for a while now) of the e2e nvidia gpus test.
And it will probably not change in the foreseeable future either.

Agreed that it would be better if it could be applied across multiple environment :)

Reply:
Yep I noticed that an hour ago when moving this to node_e2e.

> That may require us to have a fake device plugin implementation just for testing purpose. I think that will probably fall into 1.9 release timeline

I think so too

> For now, maybe we can add this test into the existing test/e2e/scheduling/nvidia-gpus.go given that it is nvidia gpu specific?

We're moving this to node_e2e as per @vishh's suggestion